src/Microsoft.PowerShell.Commands.Utility/commands/utility/Join-String.cs

```diff
@@ -166,73 +166,91 @@ protected override void EndProcessing()
+            { "; ", (TabCompletionStrings.SeparatorSemiColonSpaceToolTip, "Semi-Colon-Space") },
+            { NewLineText, (StringUtil.Format(TabCompletionStrings.SeparatorNewlineToolTip, NewLineText), "Newline") },
+            { "-", (TabCompletionStrings.SeparatorDashToolTip, "Dash") },
+            { " ", (TabCompletionStrings.SeparatorSpaceToolTip, "Space") },
+        };
 
-            void AddMatching(string completionText, string listText, string toolTip)
-            {
-                if (completionText.StartsWith(wordToComplete, StringComparison.OrdinalIgnoreCase))
-                {
-                    res.Add(new CompletionResult(completionText, listText, CompletionResultType.ParameterValue, toolTip));
-                }
-            }
+        private static string GetSeparatorToolTip(string separator)
+            => s_separatorMappings.TryGetValue(separator, out var mapping)
+                ? mapping.Tooltip
+                : separator;
 
-            AddMatching("', '", "Comma-Space", "', ' - Comma-Space");
-            AddMatching("';'", "Semi-Colon", "';'  - Semi-Colon ");
-            AddMatching("'; '", "Semi-Colon-Space", "'; ' - Semi-Colon-Space");
-            AddMatching($"\"{NewLineText}\"", "Newline", $"{NewLineText} - Newline");
-            AddMatching("','", "Comma", "','  - Comma");
-            AddMatching("'-'", "Dash", "'-'  - Dash");
-            AddMatching("' '", "Space", "' '  - Space");
-            return res;
-        }
+        private static string GetSeparatorListItemText(string separator)
+            => s_separatorMappings.TryGetValue(separator, out var mapping)
+                ? mapping.ListItemText
+                : separator;
 
-        public string NewLineText
+        /// <summary>
+        /// Returns completion results for PropertyType parameter.
+        /// </summary>
+        /// <param name="commandName">The command name.</param>
+        /// <param name="parameterName">The parameter name.</param>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="commandAst">The command AST.</param>
+        /// <param name="fakeBoundParameters">The fake bound parameters.</param>
+        /// <returns>List of Completion Results.</returns>
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
         {
-            get
+            switch (parameterName)
             {
-#if UNIX
-                return "`n";
-#else
-                return "`r`n";
-#endif
+                case "FormatString":
+                    return CompletionHelpers.GetMatchingResults(
+                        wordToComplete,
+                        possibleCompletionValues: s_formatStringValues);
+
+                case "Separator":
```

Comment:
Since we want to migrate to updated IArgumentCompleter it makes no sense to put two completers in one class. I suggest to create new class SeparatorCompleter in new PR and rename current one to FormatStringCompleter.

Reply:
Given both Separator and FormatString completer have shared code, we can do the split in this PR.

Or we can merge this in and I do the split in a follow up PR.

Reply:
In general, we will have to change this code again to switch to the updated interface. So I would just add the tests and just split the two completers without changing their code, so as not to waste time. (In addition, there are other completers that we haven't touched yet, but which we will also migrate.)

Reply:
Yeah issue here is existing completer is not as comprehensive as the current on in this PR. I have made alot of improvements to quoting logic and handling of different newlines for Unix and Windows. I think half the tests won't even work with existing code.

I dont think its a waste of time to get these changes merged in first since I am adding useful code for interface such as `ListItemTextMapping` and `IsMatch` method. 

Its less work for me to migrate to interface if these are added to CompletionHelpers class first. And even better if a completer is already relying on these new additions.

Reply:
Its probably better as well to move all existing completers in this repo first to `GetMatchingResults` since it will ensure interface when ready to migrate handles a wide variety of usage. 

And if interface is not approved then we have atleast made alot of refactoring improvements anyways.

Reply:
It can take a lot of time. I'd want we get the updated interface ready for team review as fast as possible. Then it will be in next preview and most likely in the release. Otherwise, it will be postponed for a year.